src/components/AuthorizeButton.tsx

```diff
@@ -25,27 +26,66 @@ const AuthorizeButton = ({ connection }: Props) => {
     setIsLoading(false);
   };
 
+  const authorizeConnection = async () => {
+    setIsLoading(true);
+    if (connection.oauth_grant_type === 'client_credentials') {
```

Comment:
Password oauth flow was recently added and follows the same process as client_credentials. Can you add a check? `|| connection.oauth_grant_type === 'password'`